lib/core/src/receive_swap.rs

```diff
@@ -49,46 +62,117 @@ impl ReceiveSwapStateHandler {
-          Ok(RevSwapStates::TransactionMempool
-          // The lockup tx is confirmed => try to claim
-          | RevSwapStates::TransactionConfirmed) => {
-              match receive_swap.claim_tx_id {
-                  Some(claim_tx_id) => {
-                      warn!("Claim tx for Receive Swap {id} was already broadcast: txid {claim_tx_id}")
-                  }
-                  None => {
-                      self.update_swap_info(&receive_swap.id, Pending, None)
-                          .await?;
-                      match self.claim(&receive_swap).await {
-                          Ok(_) => {}
-                          Err(err) => match err {
-                              PaymentError::AlreadyClaimed => warn!("Funds already claimed for Receive Swap {id}"),
-                              _ => error!("Claim for Receive Swap {id} failed: {err}")
-                          }
-                      }
-                  }
-              }
-              Ok(())
-          }
-
-          Ok(_) => {
-              debug!("Unhandled state for Receive Swap {id}: {swap_state}");
-              Ok(())
-          },
-
-          _ => Err(anyhow!("Invalid RevSwapState for Receive Swap {id}: {swap_state}")),
-      }
+            Ok(
+                RevSwapStates::SwapExpired
+                | RevSwapStates::InvoiceExpired
+                | RevSwapStates::TransactionFailed
+                | RevSwapStates::TransactionRefunded,
+            ) => {
+                error!("Swap {id} entered into an unrecoverable state: {swap_state:?}");
+                self.update_swap_info(id, Failed, None).await?;
+                Ok(())
+            }
+
+            // The lockup tx is in the mempool and we accept 0-conf => try to claim
+            // Execute 0-conf preconditions check
+            Ok(RevSwapStates::TransactionMempool) => {
+                if let Some(claim_tx_id) = receive_swap.claim_tx_id {
+                    warn!(
+                        "Claim tx for Receive Swap {id} was already broadcast: txid {claim_tx_id}"
+                    );
+                    return Ok(());
+                }
+
+                let boltzv2::Update::TransactionMempool { transaction, .. } = update else {
+                    return Err(anyhow!("Unexpected payload from Boltz status stream"));
+                };
+
+                let lockup_tx = utils::deserialize_tx_hex(&transaction.hex)?;
+
+                // If the amount is greater than the zero-conf limit
+                // TODO Make limit user-definable through config
+                let max_amount_sat = self.config.zero_conf_max_amount_sat();
+                let receiver_amount_sat = receive_swap.receiver_amount_sat;
+                if receiver_amount_sat > max_amount_sat {
+                    debug!("[Receive Swap {id}] Amount is too high to claim with zero-conf ({receiver_amount_sat} sat > {max_amount_sat} sat). Waiting for confirmation...");
```

Comment:
Maybe this should be an `info` or even a `warn`, as it's a very useful detail about the flow of the swap.